src/Sylius/Bundle/CoreBundle/Resources/config/services.xml

```diff
@@ -235,22 +238,62 @@
             <argument type="service" id="sylius.repository.payment_method" />
         </service>
 
-        <service id="sylius.taxation.order_shipment_taxes_by_zone_applicator" class="%sylius.taxation.order_shipment_taxes_by_zone_applicator.class%">
+        <service id="sylius.taxation.order_shipment_taxes_applicator" class="%sylius.taxation.order_shipment_taxes_applicator.class%">
             <argument type="service" id="sylius.tax_calculator" />
             <argument type="service" id="sylius.factory.adjustment" />
             <argument type="service" id="sylius.tax_rate_resolver" />
         </service>
-        <service id="sylius.taxation.order_items_taxes_by_zone_applicator" class="%sylius.taxation.order_items_taxes_by_zone_applicator.class%">
+        <service id="sylius.taxation.order_items_taxes_applicator" class="%sylius.taxation.order_items_taxes_applicator.class%">
             <argument type="service" id="sylius.tax_calculator" />
             <argument type="service" id="sylius.factory.adjustment" />
             <argument type="service" id="sylius.integer_distributor" />
             <argument type="service" id="sylius.tax_rate_resolver" />
         </service>
-        <service id="sylius.taxation.order_taxes_applicator" class="%sylius.taxation.order_taxes_applicator.class%">
+        <service id="sylius.taxation.order_item_units_taxes_applicator" class="%sylius.taxation.order_item_units_taxes_applicator.class%">
+            <argument type="service" id="sylius.tax_calculator" />
+            <argument type="service" id="sylius.factory.adjustment" />
+            <argument type="service" id="sylius.tax_rate_resolver" />
+        </service>
+
+        <service id="sylius.taxation.order_items_based_strategy" class="%sylius.taxation.composite_order_taxes_calculation_strategy.class%">
+            <argument key="type">order_items_based</argument>
+            <argument type="service">
+                <service
+                        factory-service="sylius.settings.manager"
+                        factory-method="loadSettings"
+                        class="Sylius\Bundle\SettingsBundle\Model\Settings"
+                >
```

Comment:
I'm not sure about the multi-line split, seeing as they don't stick to 80 chars in their XML, but a small detail